mealie/core/security/security.py

```diff
@@ -128,10 +129,35 @@ def user_from_ldap(db: AllRepositories, username: str, password: str) -> Private
     return user
 
 
-def authenticate_user(session, email: str, password: str) -> PrivateUser | bool:
+def _create_new_jwt_user(db: AllRepositories, claims: dict) -> PrivateUser:
+    settings = get_app_settings()
+    return db.users.create(
+        {
+            "full_name": claims[settings.JWT_AUTH_NAME_CLAIM],
+            "username": claims[settings.JWT_AUTH_USERNAME_CLAIM],
+            "email": claims[settings.JWT_AUTH_EMAIL_CLAIM],
+            "password": hash_password(secrets.token_urlsafe(13)),  # 13 char long random password
+            "group": settings.DEFAULT_GROUP,
+            "admin": True,
+        }
+    )
```

Comment:
This function is used to create users on the fly based on the information provided by the claims. The fields in which the claims come are configurable, currently the fields picked up from the claims to create a new user are:
`full_name`, `username` and `email`.
Since the password is not important, as this user won't ever login using its password I'm setting it to a random string.

Finally I left the user as admin (no strong reason for this, maybe this should be changed?), and assigned it to the default group.

This I think is a simple enough approach, I'm not sure but probably more sophisticated things could be done like, configuring groups on the IdP and match them, or pass the profile picture, but I leave it to maybe an enhancement.